astropy/units/typing_utils.py

```diff
@@ -1,116 +1,64 @@
-"""
-Utilities to help with typing units.
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
 
-Function to generate module-level stub files (.pyi) to help type checkers
-understand the dynamically created unit objects in the astropy.units subpackage.
-It inspects each specified module defining units and writes out a .pyi file
-containing the types of all discoverable UnitBase and FunctionUnitBase instances.
+"""Utilities for generating type stubs for unit definition modules.
 
 This module is meant for internal use only (in setup.py).
 """
 
-import importlib
-import unicodedata
-import warnings
-from collections.abc import Iterable, Iterator
-from pathlib import Path
-
-import astropy.units
-from astropy.units import FunctionUnitBase, UnitBase
-from astropy.utils.exceptions import AstropyDeprecationWarning
-
-# The modules that contain dynamically generated units and need stubs.
-# We do not generate stubs for modules like "core" or "quantity" because
-# type checkers can analyze their statically defined objects directly.
-MODULES_TO_STUB = [
```

Comment:
I like it here on top, and you could replace the stings with modules. But obviously not a big deal.